### PR TITLE
ENH: Add more 1.5.0 features

### PR DESCRIPTION
**Pull request by bashtage**
- [ ] Closes #xxxx (Replace xxxx with the Github issue number)
- [ ] Tests added: Please use `assert_type()` to assert the type of any return value

### CODE REVIEW

**Comment by bashtage** [comment]
More OSX timeout.

image

**Comment by bashtage** [comment]
Second time out of OSX 3.10

image

**Comment by bashtage** [comment]
image

**Comment by bashtage** [comment]
3rd attempt time out.

image

**Comment by bashtage** [comment]
Attempt 4 time out

image

**Comment by Dr-Irv** [comment]
We need to increase the time limit.  Can you modify the other PR to do that?

**Comment by bashtage** [comment]
> We need to increase the time limit. Can you modify the other PR to do that?

I'll try but I think it is stuck and any timeout won't fix it.  It should only take a couple of seconds to finish what it is doing when it stops working.

File: pandas-stubs/io/parsers/readers.pyi [hunk 1]
```diff
@@ -44,7 +48,7 @@ def read_csv(
     | npt.NDArray
     | Callable[[str], bool]
     | None = ...,
-    dtype: DtypeArg | None = ...,
+    dtype: DtypeArg | defaultdict[str, Dtype] | None = ...,
```
Comment (Dr-Irv) [inline pandas-stubs/io/parsers/readers.pyi:51]: doesn't this have to be changed for all the overloads in `read_csv()` ?

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
One small change on `read_csv()`

File: pandas-stubs/io/parsers/readers.pyi (hunk 1 above)
Comment (bashtage) [reply pandas-stubs/io/parsers/readers.pyi:51]: Probably, and also for read_clipboard.

Comment (bashtage) [reply pandas-stubs/io/parsers/readers.pyi:51]: Fixed.

File: tests/test_frame.py [hunk 2]
```diff
@@ -821,6 +821,19 @@ def test_types_to_feather() -> None:
             df.to_feather(file)
 
 
+def test_arrow_dtype() -> None:
```
Comment (Dr-Irv) [inline tests/test_frame.py:824]: this should go in `test_pandas.py` not `test_frame.py`

File: pandas-stubs/io/parsers/readers.pyi (hunk 1 above)
Comment (Dr-Irv) [reply pandas-stubs/io/parsers/readers.pyi:51]: I'm still seeing that only one of the overloads was changed.

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
1. move a test as indicated
2. Not seeing all overloads for `read_csv()` changed

**Review by Dr-Irv — APPROVED** [review]
thanks @bashtage

**Comment by Dr-Irv** [comment]
@bashtage ping when green.  PR looks good, but testing is failing.

**Comment by bashtage** [comment]
`defaultdict` typing is very problematic.  Best we can do for now seems to be just a plain `defaultdict` with no type info.  Using teh obvious choices `typing.DefautlDict[Hashable, Dtype]` of `[HashableT,Dtype]` do not work.  Even `str, Dtype` don't work.

**Review by Dr-Irv — APPROVED** [review]
thanks @bashtage

**Pull request merged**